pandas/_libs/index.pyx

```diff
@@ -681,5 +687,10 @@ cdef class BaseMultiIndexCodesEngine:
             return False
 
 
+cdef class ObjectEngine(IndexEngine):
```

Comment:
I would move this much higher up, is it the intention NOT to inherit from this?

Reply:
can you add a class description (ideally for other engines too, but ok for here now)

Reply:
Yah inheritance in this file is weird;  DatetimeEngine etc all subclass Int64Engine, which isn't defined until lower in the file once the "include ..." is done.  I'll move this up and add a description, sure.